### PR TITLE
Several minor BlockOut improvements

### PR DESCRIPTION
**Pull request by Nightenom**
# Changes proposed in this pull request:
- Fixed a textscale in buttonimage
- setView() in switchView now returns a boolean depending on whether target view was found or not

Review please

### CODE REVIEW

**Comment by Asherslab** [comment]
Might I suggest PRing into the blockout repo instead?

**Comment by Nightenom** [comment]
Yes, I can do this, will it get pushed back into this repo?

**Comment by Asherslab** [comment]
Not immediately AFAIK. @OrionDevelopment thoughts? Just PR it here?

**Comment by marchermans** [comment]
PR it in here. The blockout repo holds a Version, but really is for the new BlockOut.

File: src/main/java/com/minecolonies/blockout/controls/Scrollbar.java [hunk 1]
```diff
@@ -56,6 +62,13 @@ public Scrollbar(final ScrollingContainer container, final PaneParams params)
     {
         this(container);
         //  TODO: Parse Scrollbar-specific Params
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/blockout/controls/Scrollbar.java:64]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 61 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L567](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L567): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [TextField.java#L21](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L21): class "TextField" has 39 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [TextField.java#L331](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L331): The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [TextField.java#L331](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L331): Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [TextField.java#L380](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L380): Assign this magic number 6 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L412](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L412): Assign this magic number 255.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L412](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L412): Assign this magic number 255.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L470](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L470): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L474](https://github.com/Minecolonies/minecolonies/blob/1fc843617839022f046639a4f38102a284bb3d8e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L474): Assign this magic number 22 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 61 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L567](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L567): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [TextField.java#L21](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L21): class "TextField" has 39 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [TextField.java#L331](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L331): The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [TextField.java#L331](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L331): Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [TextField.java#L380](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L380): Assign this magic number 6 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L412](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L412): Assign this magic number 255.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L412](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L412): Assign this magic number 255.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L470](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L470): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L474](https://github.com/Minecolonies/minecolonies/blob/d47c7f7b6c5cccc95d184d4b1c43938104a97af9/src/main/java/com/minecolonies/blockout/controls/TextField.java#L474): Assign this magic number 22 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2507/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/blockout/views/SwitchView.java [hunk 2]
```diff
@@ -43,17 +43,24 @@ public void parseChildren(@NotNull final PaneParams params)
...
     }
 
-    public void setView(final String name)
+    /**
+     * Switch current view to view with id given as param
+     * 
+     * @param name id of view
+     * @return true if view of given name was found, else false
+     */
+    public boolean setView(final String name)
```
Comment (marchermans) [inline src/main/java/com/minecolonies/blockout/views/SwitchView.java:52]: Where is the return value used?

**Review by marchermans — CHANGES_REQUESTED** [review]
Some small things

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 61 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L567](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L567): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [TextField.java#L21](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L21): class "TextField" has 39 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [TextField.java#L331](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L331): The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [TextField.java#L331](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L331): Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [TextField.java#L380](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L380): Assign this magic number 6 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L412](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L412): Assign this magic number 255.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L412](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L412): Assign this magic number 255.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L470](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L470): Assign this magic number 3 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L474](https://github.com/Minecolonies/minecolonies/blob/b126e682832784846c3cc3875260b0ec84e7d31e/src/main/java/com/minecolonies/blockout/controls/TextField.java#L474): Assign this magic number 22 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/blockout/views/SwitchView.java (hunk 2 above)
Comment (Nightenom) [reply src/main/java/com/minecolonies/blockout/views/SwitchView.java:52]: In the upcoming guidebook, kinda sorry I just grabbed all changes in the blockout directory...

Comment (Nightenom) [reply src/main/java/com/minecolonies/blockout/views/SwitchView.java:52]: Maybe is can use findPaneOfTypeByID(id, SwitchView.class).getChildren().contains() but I think this more suitable for some bigger switchViews, maybe can also throw a warning into log

Comment (marchermans) [reply src/main/java/com/minecolonies/blockout/views/SwitchView.java:52]: Okey

**Pull request merged**